### PR TITLE
Bump mobx-state-tree and allow semver upgrades

### PR DESCRIPTION
**Pull request by cmdcolin**
The resources that we have are pinned to mobx-state-tree 3.14.1

While mobx-state-tree is foundational for what we do, and we should be careful about our version upgrades with it, I think it's worth allowing upgrades via semver. This also upgrades to the latest 3.15.0 release

Apparently this also makes our root package.json use 2 spaces instead of 4, like all the other package.json that we have, and which follows our editorconfig

#860

### CODE REVIEW

**Comment by cmdcolin** [comment]
Note there is a typescript change to cast getParent() results to any, this can be refined but previous versions returned any be default

**Comment by garrettjstevens** [comment]
The reason this was pinned to one version originally is because we access some MST internals (see `mst-reflection.js`) that are not guaranteed to be stable under semantic versioning. I do think we need to update MST regularly, but I'd rather keep it pinned so that updating it is an explicit action (changing the pinned version in the `package.json`s) an not something that might just happen as part of a `yarn upgrade`.

**Comment by cmdcolin** [comment]
I changed it to hardcode the version only where it is used rather than in things like peerDependencies

File: packages/jbrowse-web/src/sessionModelFactory.ts [hunk 1]
```diff
@@ -395,27 +395,27 @@ export default function sessionModelFactory(pluginManager: any) {
...
       },
 
       removeSavedSession(sessionSnapshot: any) {
-        return getParent(self).jbrowse.removeSavedSession(sessionSnapshot)
+        return getParent<any>(self).jbrowse.removeSavedSession(sessionSnapshot)
       },
 
       renameCurrentSession(sessionName: string) {
-        return getParent(self).renameCurrentSession(sessionName)
+        return getParent<any>(self).renameCurrentSession(sessionName)
```
Comment (rbuels) [inline packages/jbrowse-web/src/sessionModelFactory.ts:406]: what are all these `<any>` template args about?

Comment (cmdcolin) [reply packages/jbrowse-web/src/sessionModelFactory.ts:406]: getParent in mobx-state-tree is a templated return type. previously it had return type any. we can narrow it down to a more specific thing if there is interest in typing the rootmodel, etc.

Comment (cmdcolin) [reply packages/jbrowse-web/src/sessionModelFactory.ts:406]: that is to say, it was just changed in typescript in 3.15.0 to be unknown return type

**Pull request merged**